File: AlgorithmsProject/src/in/algorithms/decomposition/badapproach/Number.scala
package in.algorithms.decomposition.badapproach

class Number(n : Integer) extends Expr {
  def isNum = true
  def isSum = false
  def leftOp = throw new Error("leftOp of Number")
  def rightOp = throw new Error("rightOp of Number")
  def num = n
}

File: AlgorithmsProject/src/in/algorithms/decomposition/badapproach/Number.java
package in.algorithms.decomposition.badapproach;

public class Number implements Expr {
    private final int n;
    public Number(int n) { this.n = n; }

    @Override
    public boolean isNumber() { return true; }

    @Override
    public boolean isSum() { return false; }

    @Override
    public int numValue() { return n; }

    @Override
    public Expr leftOp() { throw new UnsupportedOperationException(); }

    @Override
    public Expr rightOp() { throw new UnsupportedOperationException(); }
}
